src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -523,11 +512,23 @@ def astype(self, dtype: str | type | ExtensionDtype, copy: bool = True) -> Index
             # Ensure that self.astype(self.dtype) is self
             return self.copy() if copy else self
 
+        result_class = Index
+        if dtype == "datetime64[ns]":
```

Comment:
not just this but `is_datetime64_any_dtype`

Reply:
Ack